pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -141,9 +145,126 @@ func (pl *PodTopologySpread) EventsToRegister() []framework.ClusterEventWithHint
 		// an unschedulable Pod schedulable.
 		// - Delete. An unschedulable Pod may fail due to violating an existing Pod's topology spread constraints,
 		// deleting an existing Pod may make it schedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}, QueueingHintFn: pl.isSchedulableAfterPodChange},
 		// Node add|delete|updateLabel maybe lead an topology key changed,
 		// and make these pod in scheduling schedulable or unschedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.UpdateNodeLabel}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
 	}
 }
+
+func (pl *PodTopologySpread) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPod, modifiedPod, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPodChange", "oldObj", oldObj, "newObj", newObj)
+		return framework.Queue, err
+	}
+
+	constraints, err := pl.getConstraints(pod)
+	if err != nil {
+		logger.Error(err, "get constraints from pod failed", "pod", klog.KObj(pod))
+		return framework.Queue, err
+	}
+
+	// Pod is modified. Return Queue when the label(s) matching topologySpread's selector is added, changed, or deleted.
+	if modifiedPod != nil && originalPod != nil {
+		if len(modifiedPod.Spec.NodeName) == 0 || reflect.DeepEqual(modifiedPod.Labels, originalPod.Labels) {
```

Comment:
means the same though, I prefer this. Can you modify the same in other places in this PR as well.

```suggestion
		if modifiedPod.Spec.NodeName != "" || reflect.DeepEqual(modifiedPod.Labels, originalPod.Labels) {
```

Reply:
I think you meant to say `if modifiedPod.Spec.NodeName == ""`

Reply:
Yes! 😅